pyproject.toml

```diff
@@ -240,7 +240,7 @@ exclude_also = [
     "if __name__ == .__main__.:",
     ]
 skip_covered = true
-fail_under = 90.0
+fail_under = 89.0
```

Comment:
A not acceptable change, we want to have a higher % not lower !!

Reply:
Hmm, since this code is going to be [overhauled anyways](https://github.com/pymodbus-dev/pymodbus/pull/2232#pullrequestreview-2176716262) does test coverage percentage matter all that much right now? Getting typing working better should hopefully make it easier to identify unused codepaths and do further cleanup so I'm not sure we should be actually be all that concerned about coverage at the moment for things that are being refactored like the transaction code. I'm suspecting that there's some unused code branches in the sync client transaction code.

Reply:
> You do not need to “suspect”, use “pytest —cov” and you know.

I mean, the code additions were all entirely to fix type check errors, I have no idea if some of those branches in the sync client transaction code can even be hit in practice.

> Exactly refactoring code is a good reason to have near 100% coverage before starting, because it helps detect side effects.

Ideally but seems main problem is the type checker is flagging a bunch of issues(lack of checks for `None` for example) and I wouldn't know how to write tests to hit those conditions, the sync transaction code is super messy and hard to follow, at least with type checks we can start restricting the allowable types in various args/variables more easily.

Reply:
Well maybe you have defined the wrong types, which at least seems to be a possibility.

If you change code, we need to verify why it (in your opinion) did not work earlier, and why it works now. I strongly believe that it is a sound and natural demand ! we do not do changes just because the typing you define causes problems.

That all being said, there might of  and most likely are bugs in the code, which proper typing will bring to the surface, but I am sure you understand, why proof is needed..

If it's that hard to write tests, then maybe it is also hard to write the correct typing, and not just guessing what is correct !! meaning you could always reduce the PR to the part where you know how to write tests.

Adding types to code you do not understand, it quite a mountain to climb !! it is hard enough to add typing to code you understand. Just to give you an example, we had discussions during month how to add typing to mixin.py until someone suggested templates.

Reply:
> Well maybe you have defined the wrong types, which at least seems to be a possibility.

The tests also appear to use the wrong types in a number of places which is why I was trying to fix those in #2249.

> If it's that hard to write tests, then maybe it is also hard to write the correct typing, and not just guessing what is correct !! meaning you could always reduce the PR to the part where you know how to write tests.

It's also hard to use the tests to validate typing issues when there's a lot of [typing issues being flagged by mypy](https://github.com/pymodbus-dev/pymodbus/pull/2249#issuecomment-2241202971) even on `dev`.

> Adding types to code you do not understand, it quite a mountain to climb !! it is hard enough to add typing to code you understand. Just to give you an example, we had discussions during month how to add typing to mixin.py until someone suggested templates.

If we fix the typing errors in the tests it makes it easier to validate that the types here are correct in transaction.py doesn't it?